src/Storage/Services/InstanceService.cs

```diff
@@ -13,25 +16,91 @@ namespace Altinn.Platform.Storage.Services
     public class InstanceService : IInstanceService
     {
         private readonly IInstanceRepository _instanceRepository;
+        private readonly IDataService _dataService;
+        private readonly IApplicationService _applicationService;
         private readonly IInstanceEventService _instanceEventService;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="InstanceService"/> class.
         /// </summary>
-        public InstanceService(IInstanceRepository instanceRepository, IInstanceEventService instanceEventService)
+        public InstanceService(IInstanceRepository instanceRepository, IDataService dataService, IApplicationService applicationService, IInstanceEventService instanceEventService)
         {
             _instanceRepository = instanceRepository;
+            _dataService = dataService;
+            _applicationService = applicationService;
             _instanceEventService = instanceEventService;
         }
 
         /// <inheritdoc/>
-        public async Task<SignDocument> CreateSignDocument(int instanceOwnerPartyId, Guid instanceGuid, SignRequest signRequest)
+        public async Task<(bool Created, ServiceError ServiceError)> CreateSignDocument(int instanceOwnerPartyId, Guid instanceGuid, SignRequest signRequest, int userId)
```

Comment:
We could consider moving this to a "Sign service"

Reply:
It's no problem to call it SignService but we use instanceRepository so it was clean to call it InstanceService. 

UserId is userId of the authenticated user. Users of the api can send in whatever they want in requestbody but like this we can log who did it.